### PR TITLE
Add similarity module config in index resource

### PR DESCRIPTION
**Pull request by jiahuijiang**
Add configuration to allow setting similarity config
https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules-similarity.html

### CODE REVIEW

File: es/resource_elasticsearch_index_test.go [hunk 1]
```diff
@@ -273,6 +273,19 @@ resource "elasticsearch_index" "test" {
...
+
+	testAccElasticsearchIndexWithSimilarityConfig = `
+resource "elasticsearch_index" "test_similarity_config" {
+  name = "terraform-test-update-similarity-module"
+  number_of_shards   = 1
+  number_of_replicas = 1
+  index_similarity_default = jsonencode({
+    type: "BM25",
+    b: 0.25,
+    k1: 1.2
```
Comment (phillbaker) [inline es/resource_elasticsearch_index_test.go:286]: The linter complained about HCL formatting, I think you're missing a trailing comma here

**Review by phillbaker — COMMENTED** [review]
this looks good, thanks for the contribution!

File: es/resource_elasticsearch_index_test.go (hunk 1 above)
Comment (jiahuijiang) [reply es/resource_elasticsearch_index_test.go]: @phillbaker ran linter locally and updated :)

**Comment by jiahuijiang** [comment]
@phillbaker I think there was a flake running CI :(
image

**Comment by jiahuijiang** [comment]
@phillbaker are there plans to cut a release soon? :D

**Pull request merged**